src/main/java/tars/Task.java

```diff
@@ -0,0 +1,196 @@
+package tars;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class represents a generic class with a Description and isDone status
+ * Serves as a base class for other Task types like ToDos, Deadline, Event
+ *
+ * @author csk
+ * @version 1
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for new Task with specified description
+     * The task will be initialised with marked as not done
+     *
+     * @param description (description of Task)
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a status icon depending on the isDone status of the Task
+     *
+     * @return "X" if task is done, " " otherwise returned
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Marks the task as done
+     */
+    public void mark() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as not done
+     */
+    public void unmark() {
+        this.isDone = false;
+    }
+
+    /**
+     * Returns status and description of the task
+     *
+     * @return String Representation of task
+     */
+    @Override
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
+
+    /**
+     * Returns string format of Task to be stored in the text file to retrieved later
+     *
+     * @return String Representation of task
+     */
+    public String toFileFormat() {
+        return "";
+    }
+}
+/**
+* Class represents ToDos task, which only has a description
+* Extends from Task class
+*
+* @see Task
+*/
+class ToDos extends Task {
+    /**
+     * Constructs for new ToDos Task with description stated
+     *
+     * @param description (of specified ToDos task)
+     */
+    public ToDos(String description) {
+        super(description);
+    }
+
+    /**
+     * Returns a string representation of the status and description of the task with "[T]"
+     *
+     * @return String Representation of ToDos task
+     */
+    @Override
+    public String toString() {
+        return "[T] " + super.toString();
+    }
+
+    /**
+     * Returns string format of Task to be stored in the text file to retrieved later
+     * 1 is added to string to indicate the task is done, 0 is added to indicate unmarked tasks
+     *
+     * @return String Representation of task
+     */
+    @Override
+    public String toFileFormat() {
+        if (isDone) {
+            return "T " + "1 " + description;
+        } else {
+            return "T " + "0 " + description;
+        }
+    }
+}
+/**
+ * Class represents Deadline Task, which has a specific deadline
+ * Extends from Task class and also has an extra field for deadline of task
+ *
+ * @see Task
+ */
+class Deadline extends Task {
```

Comment:
consider putting each type of task in its own file, then create a package for tasks